e2e_playwright/st_experimental_audio_input_test.py

```diff
@@ -0,0 +1,384 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import pytest
+from playwright.sync_api import FrameLocator, Locator, Page, Route, expect
+
+from e2e_playwright.conftest import IframedPage, ImageCompareFunction, wait_for_app_run
+from e2e_playwright.shared.app_utils import (
+    check_top_level_class,
+    click_button,
+    click_form_button,
+    expect_help_tooltip,
+    get_element_by_key,
+)
+
+
+def stop_recording(audio_input: Locator, app: Page):
+    """Stop recording audio and wait for the recording to complete."""
+    audio_input.get_by_role("button", name="Stop recording").click()
+    app.wait_for_timeout(5000)  # ci seems to be very slow so adding wait here
+
+
+def ensure_waveform_is_not_rendered(audio_input: Locator):
+    expect(audio_input.get_by_test_id("stAudioInputWaveSurfer")).not_to_be_visible()
+
+    time_code = audio_input.get_by_test_id("stAudioInputWaveformTimeCode")
+    expect(time_code).to_have_text("00:00")
+
+    audio_input.hover()
+    expect(
+        audio_input.get_by_role("button", name="Clear recording")
+    ).not_to_be_visible()
+
+
+def ensure_waveform_rendered(audio_input: Locator):
+    # Check for the waveform and time code
+    expect(audio_input.get_by_test_id("stAudioInputWaveSurfer")).to_be_visible()
+
+    time_code = audio_input.get_by_test_id("stAudioInputWaveformTimeCode")
+    expect(time_code).to_be_visible()
+    expect(time_code).not_to_have_text("00:00")
+
+    audio_input.hover()
+    expect(audio_input.get_by_role("button", name="Clear recording")).to_be_visible()
+    expect(audio_input.get_by_role("button", name="Download recording")).to_be_visible()
+
+
+def test_audio_input_renders(app: Page):
+    """Test that the audio input component is rendered the correct number of times."""
+    audio_input_elements = app.get_by_test_id("stAudioInput")
+    count = 7  # Expected number of audio input elements
+
+    # Verify that the expected number of elements is rendered
+    expect(audio_input_elements).to_have_count(count)
+
+    # Check each element is visible
+    for i in range(count):
+        expect(audio_input_elements.nth(i)).to_be_visible()
+
+
+def test_experimental_audio_input_renders_with_deprecation_warning(app: Page):
+    """Test that the experimental_audio_input component renders with a deprecation warning."""
+    audio_input_warning_elements = app.get_by_test_id("stAlertContainer")
+    count = 7  # Expected number of audio input elements
+
+    # Verify that the expected number of elements is rendered
+    expect(audio_input_warning_elements).to_have_count(count)
+
+    expect(app.get_by_test_id("stAlertContainer").first).to_have_text(
```

Comment:
This test is specifically checks that warning rendered for `st.experimental_audio_input`